### PR TITLE
Update version of golang to 1.18.3

### PR DESCRIPTION
**Pull request by rahulPalliyalil**

### CODE REVIEW

File: dp-hello-world-controller/go.mod [hunk 1]
```diff
@@ -1,26 +1,25 @@
...
-	github.com/smartystreets/goconvey v1.7.2
 )
 
 require (
 	github.com/BurntSushi/toml v0.3.1 // indirect
 	github.com/ONSdigital/dp-api-clients-go v1.43.0 // indirect
 	github.com/ONSdigital/dp-mongodb-in-memory v1.2.0 // indirect
-	github.com/ONSdigital/dp-net/v2 v2.0.0 // indirect
-	github.com/aws/aws-sdk-go v1.42.47 // indirect
+	github.com/ONSdigital/dp-net/v2 v2.4.0 // indirect
```
Comment (rafahop) [inline dp-hello-world-controller/go.mod:21]: When I run the command to generate the controller, this dependency shows as a direct dependency?

File: dp-hello-world-controller/go.mod [hunk 2]
```diff
@@ -1,26 +1,25 @@
...
-	github.com/ONSdigital/dp-net v1.4.1
-	github.com/ONSdigital/dp-renderer v1.21.0
+	github.com/ONSdigital/dp-net v1.5.0
+	github.com/ONSdigital/dp-renderer v1.46.1
 	github.com/ONSdigital/log.go/v2 v2.2.0
-	github.com/cucumber/godog v0.12.4
+	github.com/cucumber/godog v0.12.5
 	github.com/gorilla/mux v1.8.0
 	github.com/kelseyhightower/envconfig v1.4.0
-	github.com/smartystreets/goconvey v1.7.2
```
Comment (rafahop) [inline dp-hello-world-controller/go.mod:15]: When I run the command to generate the controller, this dependency is still there?

**Review by rafahop — COMMENTED** [review]
I get some differences when running the command for the controller, can we review it?
	- modified:   dp-hello-world-controller/ci/scripts/component.sh
	- modified:   dp-hello-world-controller/go.mod
	- modified:   dp-hello-world-controller/go.sum
	- modified:   dp-hello-world-controller/service/initialise.go

Also, would you mind updating the readme file so that the commands to regenerate these projects refer to Go 1.18.3 and include the missing command for `dp-hello-world-event`:

```
dp generate-project  --create-repository n --name "dp-hello-world-event" --description "ONS service that processes an example event" --go-version 1.18.3 --port 8123 --project-location . --type event-driven
```

File: dp-hello-world-controller/go.mod (hunk 1 above)
Comment (rahulPalliyalil) [reply dp-hello-world-controller/go.mod]: ok, reran go mod tidy and that fixed this.

**Comment by rahulPalliyalil** [comment]
> I get some differences when running the command for the controller, can we review it? - modified: dp-hello-world-controller/ci/scripts/component.sh - modified: dp-hello-world-controller/go.mod - modified: dp-hello-world-controller/go.sum - modified: dp-hello-world-controller/service/initialise.go
> 
> Also, would you mind updating the readme file so that the commands to regenerate these projects refer to Go 1.18.3 and include the missing command for `dp-hello-world-event`:
> 
> ```
> dp generate-project  --create-repository n --name "dp-hello-world-event" --description "ONS service that processes an example event" --go-version 1.18.3 --port 8123 --project-location . --type event-driven
> ```

This is done now.

File: dp-hello-world-controller/ci/scripts/component.sh [hunk 3]
```diff
@@ -1,7 +1,5 @@
 #!/bin/bash -eux
 
-export SITE_DOMAIN=localhost
-
```
[the remark this answers is not in the document]
Comment (rahulPalliyalil) [reply dp-hello-world-controller/ci/scripts/component.sh]: Done now.

**Review by rafahop — DISMISSED** [review]

**Review by rafahop — APPROVED** [review]

**Pull request merged**